arbiter-core/src/manager.rs

```diff
@@ -188,30 +193,31 @@ impl Manager {
         environment_label: S,
     ) -> Result<(), ManagerError> {
         match self.environments.get_mut(&environment_label.clone().into()) {
-            Some(environment) => match environment.state.load(std::sync::atomic::Ordering::Relaxed)
-            {
-                State::Initialization => {
-                    environment.run();
-                    info!("Started environment labeled {}", environment_label.into());
-                    Ok(())
-                }
-                State::Paused => {
-                    environment
-                        .state
-                        .store(State::Running, std::sync::atomic::Ordering::Relaxed);
-                    let (lock, pausevar) = &*environment.pausevar;
-                    let _guard = lock.lock().unwrap();
-                    pausevar.notify_all();
-                    info!("Restarted environment labeled {}", environment_label.into());
-                    Ok(())
+            Some(environment) => {
+                match environment.state.load(std::sync::atomic::Ordering::SeqCst) {
+                    State::Initialization => {
+                        environment.run();
+                        info!("Started environment labeled {}", environment_label.into());
+                        Ok(())
+                    }
+                    State::Paused => {
+                        environment
+                            .state
+                            .store(State::Running, std::sync::atomic::Ordering::SeqCst);
+                        let (lock, pausevar) = &*environment.pausevar;
+                        let _guard = lock.lock().unwrap();
+                        pausevar.notify_all();
+                        info!("Restarted environment labeled {}", environment_label.into());
+                        Ok(())
+                    }
+                    State::Running => Err(ManagerError::EnvironmentAlreadyRunning {
+                        label: environment_label.into(),
+                    }),
+                    State::Stopped => Err(ManagerError::EnvironmentStopped {
+                        label: environment_label.into(),
+                    }),
                 }
-                State::Running => Err(ManagerError::EnvironmentAlreadyRunning {
-                    label: environment_label.into(),
-                }),
-                State::Stopped => Err(ManagerError::EnvironmentStopped {
-                    label: environment_label.into(),
-                }),
-            },
+            }
             None => Err(ManagerError::EnvironmentDoesNotExist {
                 label: environment_label.into(),
```

Comment:
We can probably avoid this double match nesting which would be easier to read in the future.